frmts/georaster/cpl_vsil_ocilob.cpp

```diff
@@ -93,62 +92,101 @@ class VSIOCILobHandle : public VSIVirtualHandle
 // ****************************************************************************
 
-// ----------------------------------------------------------------------------
-//                                                          WSIOCILobFSHandle()
-// ----------------------------------------------------------------------------
-
-WSIOCILobFSHandle::WSIOCILobFSHandle()
-{
-    poStatement  = nullptr;
-    phLocator    = nullptr;
-    poConnection = nullptr;
-}
-
 // -----------------------------------------------------------------------------
-//                                                          ~WSIOCILobFSHandle()
+//                                                          ParseIdentificator()
 // -----------------------------------------------------------------------------
 
-WSIOCILobFSHandle::~WSIOCILobFSHandle()
+char** WSIOCILobFSHandle::ParseIdentificator( const char* pszFilename )
 {
-    if( phLocator )
+    if( strncmp(pszFilename, "/vsiocilob/", strlen("/vsiocilob/") ) != 0 )
     {
-        OWStatement::Free( &phLocator, 1 );
+        return nullptr;
     }
 
-    if( poStatement )
-    {
-        delete poStatement;
-    }
+    char** papszParam = CSLTokenizeString2(
+                            &pszFilename[strlen("/vsiocilob/")], ",",
+                            CSLT_HONOURSTRINGS | CSLT_ALLOWEMPTYTOKENS |
+                            CSLT_STRIPLEADSPACES | CSLT_STRIPENDSPACES );
 
-    if( poConnection )
+    if( CSLCount( papszParam ) < 6 )
     {
-        delete poConnection;
+        CSLDestroy( papszParam );
+        return nullptr;
     }
+
+    return papszParam;
 }
 
 // -----------------------------------------------------------------------------
-//                                                          ParseIdentificator()
+//                                                                        GetConnection()
 // -----------------------------------------------------------------------------
 
-char** WSIOCILobFSHandle::ParseIdentificator( const char* pszFilename )
+OWConnection* WSIOCILobFSHandle::GetConnection(char** papszParam)
 {
-    if( strncmp(pszFilename, "/vsiocilob/", strlen("/vsiocilob/") ) != 0 )
+    OWConnection *poConnection = nullptr;
+
+    if( ! papszParam )
     {
         return nullptr;
     }
 
-    char** papszParam = CSLTokenizeString2(
-                            &pszFilename[strlen("/vsiocilob/")], ",",
-                            CSLT_HONOURSTRINGS | CSLT_ALLOWEMPTYTOKENS |
-                            CSLT_STRIPLEADSPACES | CSLT_STRIPENDSPACES );
+    if( strlen( papszParam[0] ) == 0 &&
+        strlen( papszParam[1] ) == 0 &&
+        strlen( papszParam[2] ) == 0 )
+    {
+        /* In an external procedure environment, before opening any
+         * dataset, the caller must pass the with_context as an
+         * string metadata item OCI_CONTEXT_PTR to the driver. */
+
+        OCIExtProcContext* with_context = nullptr;
+
+        const char* pszContext = GDALGetMetadataItem(
```

Comment:
this doesn't make sense to me. Where does the GeoRaster driver sets a OCI_CONTEXT_PTR metadata item on itself ?

Reply:
This is set internally from the database external procedure so that user can call gdal from oracle pl/sql code.

Reply:
hum I see...